### PR TITLE
Implement for Rigel driver get_prop_range for CAP_PROP_LEAP_PULSEWIDTHMODULATION

### PR DESCRIPTION
**Pull request by jj-tetraquark**
Provides interface to requests the firmware-set safe range of the LED pulse width

### CODE REVIEW

File: src/driver_rigel.cpp [hunk 1]
```diff
@@ -138,12 +138,21 @@ class PropertyDriverRigel : public IPropertyDriver {
       case CAP_PROP_LEAP_LEDS:
         *max_val = 1;
         break;
+      case CAP_PROP_LEAP_PULSEWIDTHMODULATION: {
+        vector<uint8_t> buf(sizeof(LEAP_DEVCONFIG));
+        ok = dev_->get_xu(leap_xu_, LEAP_XU_DEVCONFIG, buf.data(), sizeof(LEAP_DEVCONFIG));
```
Comment (TomHeyUL) [inline src/driver_rigel.cpp]: Can you add a check for get_xu's return value before using buf.data() ?

Comment (jj-tetraquark) [reply src/driver_rigel.cpp]: Sure, that's now done :)

File: src/driver_rigel.cpp [hunk 2]
```diff
@@ -138,12 +138,22 @@ class PropertyDriverRigel : public IPropertyDriver {
       case CAP_PROP_LEAP_LEDS:
         *max_val = 1;
         break;
+      case CAP_PROP_LEAP_PULSEWIDTHMODULATION: {
+        vector<uint8_t> buf(sizeof(LEAP_DEVCONFIG));
```
Comment (Niram7777) [inline src/driver_rigel.cpp]: You could just declare a LEAP_DEVCONFIG instead of creating dynamic vector

File: src/driver_rigel.cpp (hunk 1 above)
Comment (Niram7777) [reply src/driver_rigel.cpp]: get_prop_range could now throw : I let you judge if 'get_xu' should be in a try-catch

Comment (jj-tetraquark) [reply src/driver_rigel.cpp]: That's a good thing to flag up but I don't think it's an issue. Not all backend have a `get_xu` that throws, and for those that do, it's due to `ioctl` failure or something else low-level. These are exceptional cases and given that this is C++ all the way up, I don't think it's right to handle the exception here.

File: src/driver_rigel.cpp (hunk 2 above)
Comment (jj-tetraquark) [reply src/driver_rigel.cpp]: You're right, that is neater. I was following the convention I'd seen in the rest of the codebase, but I prefer the way you've suggested and changed it.

**Review by Niram7777 — APPROVED** [review]

**Pull request merged**